src/parser/mod.rs

```diff
@@ -5012,17 +5018,23 @@ impl<'a> Parser<'a> {
             None
         };
 
+        let click_house_config = if dialect_of!(self is ClickHouseDialect) {
+            self.parse_optional_click_house_create_table_config()?
+        } else {
+            Default::default()
+        };
+
         let order_by = if self.parse_keywords(&[Keyword::ORDER, Keyword::BY]) {
             if self.consume_token(&Token::LParen) {
                 let columns = if self.peek_token() != Token::RParen {
-                    self.parse_comma_separated(|p| p.parse_identifier(false))?
+                    self.parse_comma_separated(|p| p.parse_expr())?
```

Comment:
can we change the type of order_by to something like [`OneOrManyWithParens<Expr>`](https://github.com/sqlparser-rs/sqlparser-rs/blob/a5b0ec183b2ca2f54627a7f14f6a0daa0eb4e2d2/src/ast/mod.rs#L824-L835)? i.e. if using parenthesis here we use the Many variant, otherwise the One variant.
I think that removes the issue with the `ORDER BY tuple(i)` vs `ORDER BY (tuple(i))` mismatch as a result?

Reply:
Updated `order_by` as an `Option<OneOrManyWithParens<Expr>>`